src/osd/OSD.cc

```diff
@@ -4756,9 +4756,9 @@ void OSD::heartbeat()
 
 bool OSD::heartbeat_reset(Connection *con)
 {
+  Mutex::Locker l(heartbeat_lock);
   auto s = con->get_priv();
   if (s) {
-    heartbeat_lock.Lock();
     if (is_stopping()) {
       heartbeat_lock.Unlock();
```

Comment:
This backport is missing a follow-on patch that fixed up this erroneous extra unlock.